### ISSUE TITLE
Downloading new model files

### ISSUE THREAD

**Issue by tumusudheer** [biemster/asr#4]
Hi,

May I know how you got the new lp_small_rnnt_gboard.zip and the other 22 files you've mentioned [here](https://hackaday.io/project/164399-android-offline-speech-recognition-natively-on-pc/log/176945-soda-speech-on-device-api#discussion-list) .

It would be great if you can tell us how to download the model.zip file you mentioned [here](https://hackaday.io/project/164399-android-offline-speech-recognition-natively-on-pc/log/169956-pixel-4-recorder-app-with-offline-transcribe). 

Thank you

**Comment by biemster** [comment]
The model zip (called lp_en_us.zip) from the Recorder app is in the res directory of the app itself. You need to extract the app with something like apktool. I'm a bit reluctant to publicly give direct links to those models since it is not my own work (and they were very likely a LOT of work), so they are also not mine to distribute..
I did however give a link in the first log of this project on hackaday.io, did you try that?
Feel free to ask further questions on the hackaday project site if you're still stuck.

**Comment by tumusudheer** [comment]
Hi @biemster ,

Thank you very much. Can you kindly let me know how can I download the Recorder app. I'm not an android user, is there a way I can download the app or apk file on my Mac book ? After that, I'll figure out how to uncompress/extract the app with some apk tool.

yeah I've downloaded the json file you've mentioned [here](https://www.gstatic.com/android/keyboard/ondevice_recognizer/superpacks-manifest-20181012.json) and able to download the single link zip file . In addition to that, I'm able to download superpacks-manifest-20190601.json and able to download new lp_rnnt-20190601.zip.

**Comment by biemster** [comment]
That's exactly the way. You can download the Recorder app from any apk mirror, I personally use apkmirror.com for such stuff.
And I did not mean some apk tool, but exactly apktool (https://ibotpeaches.github.io/Apktool/) :)
The new GBoard app loads the models from a different URL, but I'm having the impression I;m talking to you on the hackaday project page as well so you know that now already.

**Comment by tumusudheer** [comment]
In One of the discussions on hackney, I saw this link 
https://www.apkmirror.com/apk/google-inc/google-search/google-search-11-3-7-release/.
So download the apk and extracted but couldn't find any models. It would be great if you can provide the Recorder APP link in apkmirror that I can download. 

Yeah I'm talking to you on hackday as well. Can I download the GBoard app from apkmirror ? Can you provide the links for Gboard and Recorder apps apkmirror links that I can download them and I will figure out how to extract the apks using the apktool you'v mentioned.

Thank you very much.

**Comment by biemster** [comment]
https://www.apkmirror.com/apk/google-inc/google-recorder/google-recorder-1-0-271580629-release/
https://www.apkmirror.com/apk/google-inc/gboard/gboard-9-4-11-312687073-release/
or a newer one.
extract the GBoard app and search through all the files for links, so grep for json or dl.google.com and stuff like that. You'll find something for sure!

**Comment by tumusudheer** [comment]
Thanks @biemster , 

I'm able to download all lp_*.zip files from the gboard app.  Thanks once again. Also able to get the model zip (called lp_en_us.zip) from the Recorder app is in the res directory of the app itself. 

After I extracted it, I got about 26 files which are somewhat similar to the screenshot you had in the latest post with 22 files. I don't see any SODA files though. But thank you very much for your help. Now need to figure out how to run these models using TensorFlow :), the inputs and outputs of each graph :)
